primitive-types/impls/serde/src/serialize.rs

```diff
@@ -88,32 +89,27 @@ impl fmt::Display for FromHexError {
 	}
 }
 
-/// Decode given hex string into a vector of bytes.
+/// Decode given (both 0x-prefixed or not) hex string into a vector of bytes.
 ///
-/// Returns an error if the string is not prefixed with `0x`
-/// or non-hex characters are present.
+/// Returns an error if non-hex characters are present.
 pub fn from_hex(v: &str) -> Result<Vec<u8>, FromHexError> {
-	if !v.starts_with("0x") {
-		return Err(FromHexError::MissingPrefix)
-	}
+	let v = v.strip_prefix("0x").unwrap_or(v);
 
-	let mut bytes = vec![0u8; (v.len() - 1) / 2];
+	let mut bytes = vec![0u8; (v.len() + 1) / 2];
 	from_hex_raw(v, &mut bytes)?;
 	Ok(bytes)
 }
 
-/// Decode given 0x-prefixed hex string into provided slice.
+/// Decode given (both 0x-prefixed or not) hex string into provided slice.
```

Comment:
```suggestion
/// Decode given 0x-prefix-stripped hex string into provided slice.
```

Reply:
fixed

Reply:
that's not what I meant though, the string that is accepted in this function *must* be 0x-prefix-stripped, whereas other function can accept both 0x-prefixed or not hex string

Reply:
oh yeah I got it. Fixed.